### PR TITLE
update(ci): use arm64 CNCF runners for GH actions

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind cleanup

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

**What this PR does / why we need it**:

CI/build infra change

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: submodules/falcosecurity-rules
Comment (FedeDP) [inline submodules/falcosecurity-rules:1]: Can we revert this?

Comment (FedeDP) [reply submodules/falcosecurity-rules:1]: :rofl:

Comment (LucaGuerra) [reply submodules/falcosecurity-rules:1]: 🫠

**Comment by FedeDP** [comment]
CI is weirdly failing during cpack :/

**Review by leogr — APPROVED** [review]

File: .github/workflows/master.yaml [hunk 1]
```diff
@@ -42,15 +42,17 @@ jobs:
...
+  # test-dev-packages-arm64:
+  #   needs: [fetch-version, build-dev-packages-arm64]
+  #   uses: ./.github/workflows/reusable_test_packages.yaml
+  #   with:
+  #     arch: aarch64
+  #     version: ${{ needs.fetch-version.outputs.version }}
 
   publish-dev-packages:
-    needs: [fetch-version, test-dev-packages, test-dev-packages-arm64]
+    needs: [fetch-version, build-dev-packages-arm64]
```
Comment (FedeDP) [inline .github/workflows/master.yaml:55]:
```suggestion
    needs: [fetch-version, test-dev-packages, build-dev-packages-arm64]
```

**Review by FedeDP — DISMISSED** [review]
/approve

File: .github/workflows/reusable_test_packages.yaml [hunk 2]
```diff
@@ -57,7 +57,7 @@ jobs:
       - name: Run tests
         env:
           LSAN_OPTIONS: "intercept_tls_get_addr=0"
-        uses: falcosecurity/testing@main 
+        uses: falcosecurity/testing@92638104b7f37629d3228196213a0eb679906b62
```
Comment (FedeDP) [inline .github/workflows/reusable_test_packages.yaml:60]: Can we revert back to main?

Comment (LucaGuerra) [reply .github/workflows/reusable_test_packages.yaml]: done

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]

**Pull request merged**